### PR TITLE
ci: Use C# 10 inside the Uno solution

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Build or CI related changes

## What is the new behavior?

- Move to C# 10 for all Uno solution projects, including macOS ones
- Use .NET 6 wherever possible
- Use stable VS2022 for VSIX creation
- Use hosted pool for skia tests (latest hosted agents seem to introduce some memory corruption)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj [hunk 1]
```diff
@@ -39,6 +39,9 @@
 		<PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
 	</ItemGroup>
 
+	<!-- -->
```
Comment (carldebilly) [inline src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj:42]: I like your quintessential comment ;-)

File: src/Uno.WinUIRevert/Uno.WinUIRevert.csproj [hunk 2]
```diff
@@ -3,6 +3,9 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net5.0</TargetFramework>
+
+		<!-- Required for non-net6 CI build -->
+		<LangVersion>9.0</LangVersion>
   </PropertyGroup>
```
Comment (carldebilly) [inline src/Uno.WinUIRevert/Uno.WinUIRevert.csproj:9]: Weird identation

**Review by carldebilly — APPROVED** [review]

File: src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI.RemoteControl.VS/Uno.UI.RemoteControl.VS.csproj:42]: Will fix in later build!

File: src/Uno.WinUIRevert/Uno.WinUIRevert.csproj (hunk 2 above)
Comment (jeromelaban) [reply src/Uno.WinUIRevert/Uno.WinUIRevert.csproj:9]: Will fix in later build!

**Review by dr1rrb — APPROVED** [review]

**Pull request merged**